src/lib/operators/get_table.cpp

```diff
@@ -12,84 +12,176 @@
+  const auto stored_table = StorageManager::get().get_table(_name);
+
   const auto separator = description_mode == DescriptionMode::MultiLine ? "\n" : " ";
+
   std::stringstream stream;
+
   stream << name() << separator << "(" << table_name() << ")";
-  if (!_excluded_chunk_ids.empty()) {
-    stream << separator << "(" << _excluded_chunk_ids.size() << " Chunks pruned)";
-  }
+
+  stream << separator << "pruned:" << separator;
+  stream << _pruned_chunk_ids.size() << "/" << stored_table->chunk_count() << " chunk(s)";
+  if (description_mode == DescriptionMode::SingleLine) stream << ",";
+  stream << separator << _pruned_column_ids.size() << "/" << stored_table->column_count() << " column(s)";
+
   return stream.str();
 }
 
 const std::string& GetTable::table_name() const { return _name; }
 
-void GetTable::set_excluded_chunk_ids(const std::vector<ChunkID>& excluded_chunk_ids) {
-  _excluded_chunk_ids = excluded_chunk_ids;
-}
+const std::vector<ChunkID>& GetTable::pruned_chunk_ids() const { return _pruned_chunk_ids; }
+
+const std::vector<ColumnID>& GetTable::pruned_column_ids() const { return _pruned_column_ids; }
 
 std::shared_ptr<AbstractOperator> GetTable::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  auto copy = std::make_shared<GetTable>(_name);
-  copy->set_excluded_chunk_ids(_excluded_chunk_ids);
-  return copy;
+  return std::make_shared<GetTable>(_name, _pruned_chunk_ids, _pruned_column_ids);
 }
 
 void GetTable::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> GetTable::_on_execute() {
+  const auto stored_table = StorageManager::get().get_table(_name);
+
+  /**
+   * Build a sorted vector physically or logically deleted ChunkIDs (excluding those pruned, so that _pruned_chunk_ids
+   * and deleted_chunk_ids are mutually exclusive)
+   */
   DebugAssert(!transaction_context_is_set() || transaction_context()->phase() == TransactionPhase::Active,
               "Transaction is not active anymore.");
-
-  auto original_table = StorageManager::get().get_table(_name);
-  auto temp_excluded_chunk_ids = std::vector<ChunkID>(_excluded_chunk_ids);
-
   if (HYRISE_DEBUG && !transaction_context_is_set()) {
-    for (ChunkID chunk_id{0}; chunk_id < original_table->chunk_count(); ++chunk_id) {
-      DebugAssert(original_table->get_chunk(chunk_id) && !original_table->get_chunk(chunk_id)->get_cleanup_commit_id(),
+    for (ChunkID chunk_id{0}; chunk_id < stored_table->chunk_count(); ++chunk_id) {
+      DebugAssert(stored_table->get_chunk(chunk_id) && !stored_table->get_chunk(chunk_id)->get_cleanup_commit_id(),
                   "For tables with physically deleted chunks, the transaction context must be set.");
     }
   }
 
-  // Add logically & physically deleted chunks to list of excluded chunks
+  auto deleted_chunk_ids = std::vector<ChunkID>{};
```

Comment:
The comment here was helpful. With pruning of chunks without a result and excluding chunks that were logically or physically deleted, we should make the distinction clear.

Reply:
Further up there is a block comment
```c++
  /**
   * Build a sorted vector (`excluded_chunk_ids`) of physically/logically deleted and pruned ChunkIDs
   */ 
```